### PR TITLE
feat: return the url of the worker file

### LINKED ISSUE #226: [Feature Request]: Why does not the loader return the worker path?
In order to support any kinds of existing workers such as WebWorker, ServiceWorker, SharedWorker and so on. Have you thought about return path which works like file-loader or url-loader? And then let the developers to decide which type of worker should use. It indirectly solve the service worker scenario under worker-loader

### PR DESCRIPTION
**Pull request by wood1986**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [x] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

fix #226 and #11. With this change, developers can decide which Worker they want to use. Currently, Worker is allowed. What about [ServiceWorker](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker) or [SharedWorker](https://developer.mozilla.org/en-US/docs/Web/API/SharedWorker)

I want to have a discussion first before writing tests for url

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -38,7 +39,7 @@ export function pitch(request) {
 
   const filename = loaderUtils.interpolateName(
     this,
-    options.name || '[hash].worker.js',
+    options.name || this._compilation.outputOptions.filename,
```
Comment (wood1986) [inline src/index.js]: By default this should from https://webpack.js.org/configuration/output/#outputfilename

File: src/index.js [hunk 2]
```diff
@@ -49,7 +50,7 @@ export function pitch(request) {
 
   worker.options = {
     filename,
-    chunkFilename: `[id].${filename}`,
+    chunkFilename: this._compilation.outputOptions.chunkFilename,
```
Comment (wood1986) [inline src/index.js]: By default this should come from https://webpack.js.org/configuration/output/#outputchunkfilename

File: src/index.js [hunk 3]
```diff
@@ -66,9 +67,11 @@ export function pitch(request) {
     new NodeTargetPlugin().apply(worker.compiler);
   }
 
-  new SingleEntryPlugin(this.context, `!!${request}`, 'main').apply(
-    worker.compiler
-  );
+  new SingleEntryPlugin(
+    this.context,
+    `!!${request}`,
+    path.parse(options.name || this.resourcePath).name
```
Comment (wood1986) [inline src/index.js]: You should name the chunk instead of using a stable name.

**Comment by wood1986** [comment]
Any updates?

**Comment by colelawrence** [comment]
@wood1986 Nice work! I wonder if anyone is taking care of this repo atm.

**Comment by CntChen** [comment]
Nice work. This MR is exactly what I needed.
Anyone merge it into master.

**Comment by evilebottnawi** [comment]
Please read https://github.com/webpack-contrib/worker-loader/issues/226#issuecomment-657617805

**Pull request closed without merging**